### PR TITLE
chore: update k8s api version info

### PR DESCRIPTION
**Pull request by philoserf**

### CODE REVIEW

**Comment by philoserf** [comment]
no preference of bare `v1` over `core/v1` @mikeln

**Comment by philoserf** [comment]
Could this small changes have caused the CI/CD error?

```
Error: release eventrouter-jenkins-pipeline-eventrouter-pr-19 failed: Deployment.apps
"eventrouter" is invalid: [spec.selector: Required value, spec.template.metadata.labels:
Invalid value: map[string]string{"component":"eventrouter", "tier":"control-plane-addons"}:
`selector` does not match template `labels`]
```

**Comment by philoserf** [comment]
Must be a k8s 1.9 vs later api version issue.

Digging in

**Comment by philoserf** [comment]
Well, that isn't it. Deployments `apps/v1` was introduced in v1.9.

**Comment by DStorck** [comment]
there were some changes with the apiVersion when it went to apps.v1 - selectors are now immutable, this actually tried to bite me on the CKA....anyway, start there

**Comment by philoserf** [comment]
Sure enough, a selector is required in `apps/v1`. I added one here. Not certain if that is the final selector for this pull request.

File: charts/eventrouter/templates/deployment.yaml [hunk 1]
```diff
@@ -10,6 +10,9 @@ metadata:
     heritage: "{{.Release.Service}}"
 spec:
   replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      component: eventrouter
```
Comment (mikeln) [inline charts/eventrouter/templates/deployment.yaml]: I suspect this should not be hard coded.   probably should be {{ .Values.name }} ??
What is this specifying/connecting exactly?

Comment (mikeln) [reply charts/eventrouter/templates/deployment.yaml]: Ok, so matchLabels.component has to match template.metadata.labels.component
`
spec:
  replicas: {{ .Values.replicaCount }}
  selector:
    matchLabels:
      component: eventrouter
  template:
    metadata:
      labels:
        component: eventrouter
        tier: control-plane-addons
    spec:
'

Comment (mikeln) [reply charts/eventrouter/templates/deployment.yaml]: I wonder if we should really be using {{ .Values.name }} for those 2 values.

Comment (philoserf) [reply charts/eventrouter/templates/deployment.yaml]: That’s what I set in the pull request.

**Comment by philoserf** [comment]
@mikeln are your concerns resolved?

**Pull request merged**